### ISSUE TITLE
Data not sorted when onSortingChange handler is passed 

### ISSUE THREAD

**Issue by tjphopkins** [tannerlinsley/react-table#209]
## Problem Description
If an onSortingChange handler is passed to the ReactTable component, when a header is clicked to sort by column, the onSortingChange handler is called, but the data is not sorted.

This was reported as a side-issue in the closed [issue 163](https://github.com/tannerlinsley/react-table/issues/163) but I have found the [fix](https://github.com/tannerlinsley/react-table/pull/166) to not solve this problem.
 
## Steps to Reproduce
1. Pass ReactTable an onSortingChange prop
```
return (
      <ReactTable
        data={data}
        columns={columns}
        onSortingChange={() => {console.log("Sorting changed");}}
      />
)
```
2. Click on a column header to sort
3. Observe the onSortingChange handler being called and the table data not being sorted

## System Information
* **Browser & Browser Version: Chrome 57.0.2987.133**
* **Node Version: v6.1.0**
* **OS Version: OS X Yosemite 10.10.5**
* **NPM Version: 3.8.6**

**Comment by tannerlinsley** [comment]
TL;DR - The docs need to be more clear. I'll get on this asap.

This is by design actually, and maybe we need to update the documentation on this one.  `onSortingChange` and other related `onThingChange` props are used explicitly for fully controlling the component.  So when you pass `onSortingChange`, the table is expecting you to take the value in that method, store it (or optionally modify it) to your own state (local state, redux, etc), and then reinject it into your table instance in the `sorting` prop. This allows you to hold the sorting model in your own state, rather than it being trapped in the internal state of the Table.  This is especially useful if you are using a higher-order state management system like Redux.  In which case you would be able to maintain the table state between hot reloads and do a lot of other cool things.
